src/algebra/field/basic.lean

```diff
@@ -50,23 +50,59 @@ set_option old_structure_cmd true
 universe u
 variables {α β K : Type*}
 
+/-- The default definition of the coercion `(↑(a : ℚ) : K)` for a division ring `K`
+is defined as `(a / b : K) = (a : K) * (b : K)⁻¹`.
+Use `coe` instead of `of_rat_rec` for better definitional behaviour.
+-/
+def of_rat_rec [has_lift_t ℕ K] [has_lift_t ℤ K] [has_mul K] [has_inv K] : ℚ → K
+| ⟨a, b, _, _⟩ := ↑a * (↑b)⁻¹
```

Comment:
Is the irreducibility conferred by defining this with pattern matching intentional and/or beneficial? Not a big deal if you didn't think about it, but if you already did an explanatory comment would be great.

Reply:
I think either way would work. The main reason I wrote it like this is to match the inductive definitions (that have now become) [nat.unary_cast](https://leanprover-community.github.io/mathlib_docs/data/nat/cast/defs.html#nat.unary_cast) and [int.cast_def](https://leanprover-community.github.io/mathlib_docs/data/int/cast/defs.html#int.cast_def), and I didn't see much reason to change definitional behaviour: you can use `rat.cast_def` instead, which as a bonus works for all division rings.